lib/assembly/genservers/build.ex

```diff
@@ -104,6 +116,12 @@ defmodule Assembly.GenServers.Build do
   @impl true
   def handle_info(:update_status, state), do: {:noreply, state}
 
+  defp build_demand_response(options) do
+    Enum.reduce(options, %{}, fn o, map ->
+      AdditiveMap.add(map, o.component_id, o.quantity)
```

Comment:
I really like that this was extracted.

I wonder if we could accomplish the component_id -> quantity mapping without using AdditiveMap. Of course, it's not clear if there could be multiple entries for the same component in options, in which case using AdditiveMap would be correct.